### PR TITLE
Browser manager as class

### PR DESCRIPTION
**Pull request by vringar**
This PR refactors the BrowserManager to be a class inheriting from multiprocess.Process.
This change allows the logic of BrowserManager to be broken up into multiple pieces as indicated by `_start_extension`

### CODE REVIEW

File: openwpm/browser_manager.py [hunk 1]
```diff
@@ -137,7 +135,8 @@ def launch_browser_manager(self):
         unsuccessful_spawns = 0
         success = False
 
-        def check_queue(launch_status):
+        def check_queue(launch_status: Dict[Any, Any]) -> Any:
```
Comment (boolean5) [inline openwpm/browser_manager.py:138]: `launch_status` could be `Dict[str, bool]`, right?

File: openwpm/browser_manager.py [hunk 2]
```diff
@@ -156,19 +155,19 @@ def check_queue(launch_status):
...
                 self.manager_params,
                 crash_recovery,
             )
-            self.browser_manager = Process(target=BrowserManager, args=args)
             self.browser_manager.daemon = True
             self.browser_manager.start()
 
             # Read success status of browser manager
-            launch_status = dict()
+            launch_status: Dict[Any, Any] = dict()
```
Comment (boolean5) [inline openwpm/browser_manager.py:170]: Same here

File: openwpm/browser_manager.py [hunk 3]
```diff
@@ -612,130 +611,158 @@ def shutdown_browser(self, during_init: bool, force: bool = False) -> None:
...
+            self.logger.error(
+                "BROWSER %i: Crash in driver, restarting browser manager"
+                % self.browser_params.browser_id,
+                exc_info=True,
+                extra=extra,
+            )
+            self.status_queue.put(("FAILED", pickle.dumps(sys.exc_info())))
+        finally:
+            if display is not None:
+                display.stop()
```
Comment (boolean5) [inline openwpm/browser_manager.py:767]: The only difference I see from before is that now that we don't use the [`Process` wrapper](https://github.com/mozilla/OpenWPM/blob/cc10baebee4a0f3fbec38eeac17586448c903e60/openwpm/utilities/multiprocess_utils.py#L35L50) from `utilities.multiprocess_utils` if an exception is raised here it won't be logged. But I don't know if that's really a problem.

**Review by boolean5 — APPROVED** [review]
Looks good! Feel free to merge after taking a look at the comments.

File: openwpm/browser_manager.py (hunk 3 above)
Comment (vringar) [reply openwpm/browser_manager.py:767]: Yeah, that was my thinking as well. 
I think it's highly unlikely that the display stopping will fail and all other exceptions are caught in the except block above.

**Pull request merged**